File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnLookupDatabaseInfo.cs

namespace GracenoteSDK {

public class GnLookupDatabaseInfo : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnLookupDatabaseInfo(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnLookupDatabaseInfo_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnLookupDatabaseInfo obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnLookupDatabaseInfo() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnLookupDatabaseInfo(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

  public GnLookupDatabaseInfo() : this(gnsdk_csharp_marshalPINVOKE.new_GnLookupDatabaseInfo(), true) {
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnLookupDatabaseInfo.java

package com.gracenote.gnsdk;

public class GnLookupDatabaseInfo extends GnDataObject {
  private transient long swigCPtr;

  protected GnLookupDatabaseInfo(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnLookupDatabaseInfo_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnLookupDatabaseInfo obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnLookupDatabaseInfo(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnLookupDatabaseInfo() {
    this(gnsdk_javaJNI.new_GnLookupDatabaseInfo(), true);
  }

}
